protocol/pubsub/v2/protocol_test.go

```diff
@@ -20,6 +23,32 @@ type testPubsubClient struct {
 	conn *grpc.ClientConn
 }
 
+func (pc *testPubsubClient) NewWithAttributesInterceptor(ctx context.Context, projectID, orderingKey string) (*pubsub.Client, error) {
+	pc.srv = pstest.NewServer()
+	conn, err := grpc.Dial(pc.srv.Addr, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithUnaryInterceptor(customAttributesInterceptor(map[string]string{
+		"Content-Type":        "text/json",
+		"ce-dataschema":       "http://example.com/schema",
+		"ce-exbinary":         "AAECAw==",
+		"ce-exbool":           "true",
+		"ce-exint":            "42",
+		"ce-exstring":         "exstring",
+		"ce-extime":           "2020-03-21T12:34:56.78Z",
+		"ce-exurl":            "http://example.com/source",
+		"ce-id":               "full-event",
+		"ce-source":           "http://example.com/source",
+		"ce-specversion":      "1.0",
+		"ce-subject":          "topic",
+		"ce-time":             "2020-03-21T12:34:56.78Z",
+		"ce-type":             "com.example.FullEvent",
+		"Proxy-Authorization": "YWxhZGRpbjpvcGVuc2VzYW1l",
```

Comment:
So, just to be clear... you're expecting people to use `WithCustomAttributes` to set non-CE extension attributes and then `event.SetExtension` to set

Reply:
Yeah, similar to people currently using `WithCustomHeader` to set non-CE extension attributes and then `event.SetExtension` to set CE extension attributes (for HTTP).

Reply:
ok thanks. In that case, would it be possible to add this to our docs someplace? That might not be a fair ask since I think the docs are kind of lacking, but if you know of a good spot for us to describe this new Option then it would be nice to add it there. The main reason I'm asking is because any example associated with the docs should only show attribute w/o `ce-` unlike your test. I'm not keen on people thinking this is the right way to set CE attributes.

Maybe at a minimum, add some more text here: https://github.com/cloudevents/sdk-go/pull/976/files#diff-42e7fdd259e37dfd3740b150824fabc99fc50cbcb2d934ed1ca675d733121719R23 with something about how this is for protocol specific attributes and NOT for CE extensions, which should be set via the SetExtention func. Maybe?

Reply:
Done.

Reply:
thanks!